src/LiveComponent/src/EventListener/LiveComponentSubscriber.php

```diff
@@ -148,6 +152,31 @@ public function onKernelController(ControllerEvent $event): void
 
         $request->attributes->set('_mounted_component', $mounted);
 
+        // autowire live file arguments
+        if ($request->files->count()) {
+            $allFiles = $request->files->all();
+            $accessor = PropertyAccess::createPropertyAccessor();
+            foreach (LiveFileArg::liveFileArgs($component, $action) as $parameter => $fileArg) {
+                $path = $fileArg->getPropertyPath();
+
+                if (!$accessor->isReadable($allFiles, $path)) {
```

Comment:
The property path stuff is handling cases where the name of the file input is of the form `name="form[photo]"`, correct?

Reply:
Right, I didn't find simple way to translate HTML paths to property accesor path, so this came out :)

Reply:
I have decided to not support that case as we can use data-model="friendlyName" and keep type="form[data][file]" for form compatibility. It simplifies code a bit without sacrificing actual usability